astropy/convolution/core.py

```diff
@@ -243,6 +243,10 @@ def __init__(self, model=None, x_size=None, array=None, **kwargs):
 
         super().__init__(array)
 
+    @property
+    def ndim(self):
```

Comment:
Do you think it is cleaner to return `self._array.ndim` in the base class, like `shape`?

Reply:
Yes, using `self._array.ndim` would seem cleaner. This is also used in the `dimension` property. I think it makes sense to go with it here too. More generally, I think it makes sense to add this property, but perhaps with a docstring that states it is for compatibility with `np.array`, something like """Kernal dimension. Alias to dimension, for compatibility with arrays.""" (unless we want to deprecate `dimension` in favour of `ndim`).

Reply:
This approach has turned into a dead-end for addressing the larger issue. It now fails with:
```python
/home/docs/checkouts/readthedocs.org/user_builds/astropy/envs/13205/lib/python3.9/site-packages/astropy/convolution/kernels.py:docstring of astropy.convolution.kernels.Box1DKernel:56: WARNING: Exception occurred in plotting astropy-convolution-Box1DKernel-1
 from /home/docs/checkouts/readthedocs.org/user_builds/astropy/checkouts/13205/docs/api/astropy.convolution.Box1DKernel.rst:
Traceback (most recent call last):
  File "/home/docs/checkouts/readthedocs.org/user_builds/astropy/envs/13205/lib/python3.9/site-packages/matplotlib/sphinxext/plot_directive.py", line 517, in _run_code
    exec(code, ns)
  File "<string>", line 4, in <module>
  File "/home/docs/checkouts/readthedocs.org/user_builds/astropy/envs/13205/lib/python3.9/site-packages/matplotlib/pyplot.py", line 2769, in plot
    return gca().plot(
  File "/home/docs/checkouts/readthedocs.org/user_builds/astropy/envs/13205/lib/python3.9/site-packages/matplotlib/axes/_axes.py", line 1632, in plot
    lines = [*self._get_lines(*args, data=data, **kwargs)]
  File "/home/docs/checkouts/readthedocs.org/user_builds/astropy/envs/13205/lib/python3.9/site-packages/matplotlib/axes/_base.py", line 312, in __call__
    yield from self._plot_args(this, kwargs)
  File "/home/docs/checkouts/readthedocs.org/user_builds/astropy/envs/13205/lib/python3.9/site-packages/matplotlib/axes/_base.py", line 506, in _plot_args
    y = y[:, np.newaxis]
TypeError: 'Box1DKernel' object is not subscriptable
```

Any suggestions?

Reply:
So, clearly matplotlib is no longer coercing the input to arrays. Maybe then the best thing is to adjust the actual examples (i.e., use `kernel.array` or whatever it was), and add an entry to API changes or so. The only alternative is to behave even more like an array, but then the overrides of `__mul__` to mean convolution, etc., are becoming quite weird.